system/jlib/jfile.cpp

```diff
@@ -2134,16 +2153,25 @@ size32_t CFileIO::write(offset_t pos, size32_t len, const void * data)
         throw makeErrnoException(errno, "CFileIO::write");
     if (ret<len)
         throw makeOsException(DISK_FULL_EXCEPTION_CODE, "CFileIO::write");
-    if ( (extraFlags & IFEnocache) && (ret > 0) )
+    if ( (extraFlags & (IFEnocache | IFEsync)) && (ret > 0) )
```

Comment:
trivial: feels it would be slightly clearer if (ret > 0) was 1st in this expression (because other if's prior to this are based ret)